autotest/test_gwt_ssm06fmi.py

```diff
@@ -0,0 +1,429 @@
+
+    dis = flopy.mf6.ModflowGwtdis(
+        gwt,
+        nlay=nlay,
+        nrow=nrow,
+        ncol=ncol,
+        delr=delr,
+        delc=delc,
+        top=top,
+        botm=botm,
+    )
+    ic = flopy.mf6.ModflowGwtic(gwt, strt=100.0)
+    sto = flopy.mf6.ModflowGwtmst(gwt, porosity=0.3)
+    adv = flopy.mf6.ModflowGwtadv(gwt, scheme="TVD")
+    dsp = flopy.mf6.ModflowGwtdsp(gwt, alh=20.0, ath1=2, atv=0.2)
+
+    # Create the ssm sources block information
+    sourcerecarray = []
+    sourcerecarray += [("WEL-1", "AUX", "CONCENTRATION")]
+    sourcerecarray += [
+        (f"GHB-{i+1}", "AUX", "CONCENTRATION") for i in [0, 1, 2, 3]
+    ]
+    sourcerecarray += [
+        (f"DRN-{i+1}", "AUX", "CONCENTRATION") for i in [0, 1, 2]
+    ]
+
+    ssm = flopy.mf6.ModflowGwtssm(
+        gwt, print_flows=True, sources=sourcerecarray
+    )
+
+    pd = [
+        ("GWFHEAD", "../flow/flow.hds", None),
+        ("GWFBUDGET", "../flow/flow.bud", None),
+        ("GWFMOVER", "../flow/flow.mvr.bud", None),
+        ("SFR-1", "../flow/flow.sfr.bud", None),
+    ]
+    fmi = flopy.mf6.ModflowGwtfmi(
+        gwt, packagedata=pd, flow_imbalance_correction=True
+    )
+
+    sftpkdat = []
+    for irno in range(len(sfrcells)):
+        t = (irno, 1.0)
+        sftpkdat.append(t)
+
+    sftspd = {0: [[0, "INFLOW", 1.0]]}
+    sft = flopy.mf6.modflow.ModflowGwtsft(
+        gwt,
+        boundnames=False,
+        flow_package_name="SFR-1",
+        print_concentration=True,
+        save_flows=True,
+        concentration_filerecord=gwtname + ".sft.bin",
+        budget_filerecord=gwtname + ".sft.bud",
+        packagedata=sftpkdat,
+        reachperioddata=sftspd,
+        pname="SFT-1",
+    )
+
+    mvt = flopy.mf6.modflow.ModflowGwtmvt(
+        gwt,
+        pname="MVT-1",
+    )
+
+    oc = flopy.mf6.ModflowGwtoc(
+        gwt,
+        budget_filerecord=f"{gwtname}.cbc",
+        budgetcsv_filerecord=f"{gwtname}.cbc.csv",
+        concentration_filerecord=f"{gwtname}.ucn",
+        concentrationprintrecord=[
+            ("COLUMNS", ncol, "WIDTH", 15, "DIGITS", 6, "GENERAL")
+        ],
+        saverecord=[("CONCENTRATION", "ALL"), ("BUDGET", "ALL")],
+        printrecord=[("CONCENTRATION", "ALL"), ("BUDGET", "ALL")],
+    )
+
+    sim.write_simulation()
+    success, buff = sim.run_simulation(silent=False)
+    errmsg = "transport model did not terminate successfully\n{}".format(buff)
+    assert success, errmsg
+
+    # ensure budget table can be parsed
+    fname = gwtname + ".lst"
+    fname = os.path.join(wst, fname)
+    budl = flopy.utils.Mf6ListBudget(
+        fname, budgetkey="MASS BUDGET FOR ENTIRE MODEL"
+    )
+    d0 = budl.get_budget()[0]
+
+    # Load the csv representation of the budget for confirming that the model ran
+    fname = f"{gwtname}.cbc.csv"
+    fname = os.path.join(wst, fname)
+    d0 = np.genfromtxt(fname, names=True, delimiter=",", deletechars="")
+    print(d0.dtype.names)
+    return
+
+
+def test_ssm01fmi():
```

Comment:
this should be renamed to test_ssm06fmi, which is what pytest will echo to the screen

Reply:
resolved with c4a2a03